sycl/test/hier_par/hier_par_basic.cpp

```diff
@@ -0,0 +1,183 @@
+//==- hier_par_basic.cpp --- hierarchical parallelism API test -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// RUN: %clang -std=c++11 -fsycl %s -o %t.out -lOpenCL -lstdc++
+// RUN: %t.out
+
+// This test checks hierarchical parallelism invocation APIs, but without any
+// data or code with side-effects between the work group and work item scopes.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <memory>
+
+using namespace cl::sycl;
+
+template <typename GoldFnTy>
+bool verify(int testcase, int range_length, int *ptr, GoldFnTy get_gold) {
+  int err_cnt = 0;
+
+  for (int i = 0; i < range_length; i++) {
+    int gold = get_gold(i);
+
+    if (ptr[i] != gold) {
+      if (++err_cnt < 20) {
+        std::cout << testcase << " - ERROR at " << i << ": " << ptr[i]
+                  << " != " << gold << "(expected)\n";
+      }
+    }
+  }
+  if (err_cnt > 0)
+    std::cout << "-- Failure rate: " << err_cnt << "/" << range_length << "("
+              << err_cnt / (float)range_length * 100.f << "%)\n";
+  return err_cnt == 0;
+}
+
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
```

Comment:
`std::min`

Reply:
That caused some problems with compilation, which I preferred to workaround at that moment. Will see if I can make std::min work.

Reply:
Perhaps `std::min` was not working because it is `constexpr` only since C++14...
If so, `cl::sycl::min` might have the same limitation...

Reply:
I'm not sure what is the problem, but maybe it was caused by the headers we included on the device and which was removed by https://github.com/intel/llvm/pull/229.
@kbobrovs, could you rebase you patch to the latest version the branch and check if you still have issues with using `std::min`, please?

Reply:
Yes, will try.

Reply:
cl::sycl::min worked fine - this seems to be the right choice here, thanks for the suggestion